plugins/pulp_docker/plugins/importers/v1_sync.py

```diff
@@ -218,88 +203,61 @@ def find_and_read_ancestry_file(image_id, parent_dir):
             return json.load(ancestry_file)
 
 
-class GetLocalImagesStep(GetLocalUnitsStep):
-    def _dict_to_unit(self, unit_dict):
+class SaveImages(SaveUnitsStep):
+    def __init__(self, step_type=constants.SYNC_STEP_SAVE):
         """
-        convert a unit dictionary (a flat dict that has all unit key, metadata,
-        etc. keys at the root level) into a Unit object. This requires knowing
-        not just what fields are part of the unit key, but also how to derive
-        the storage path.
-        Any keys in the "metadata" dict on the returned unit will overwrite the
-        corresponding values that are currently saved in the unit's metadata. In
-        this case, we pass an empty dict, because we don't want to make changes.
-        :param unit_dict:   a flat dictionary that has all unit key, metadata,
-                            etc. keys at the root level, representing a unit
-                            in pulp
-        :type  unit_dict:   dict
-        :return:    a unit instance
-        :rtype:     pulp.plugins.model.Unit
+        Initialize the SaveImages Step, setting its type and description.
         """
-        model = models.Image(unit_dict['image_id'], unit_dict.get('parent_id'),
-                             unit_dict.get('size'))
-        return self.get_conduit().init_unit(model.TYPE_ID, model.unit_key, {},
-                                            model.relative_path)
-
+        super(SaveImages, self).__init__(step_type=step_type)
+        self.description = _('Saving images and tags')
 
-class SaveUnits(PluginStep):
-    def __init__(self, working_dir):
+    def get_iterator(self):
         """
-        :param working_dir: full path to the directory into which image files
-                            are downloaded. This directory should contain one
-                            directory for each docker image, with the ID of the
-                            docker image as its name.
-        :type  working_dir: basestring
+        This method returns an iterator that traverses the list of Images that were downloaded.
+
+        :return: a list or other iterable
+        :rtype:  iterator of pulp_docker.plugins.models.Image
         """
-        super(SaveUnits, self).__init__(step_type=constants.SYNC_STEP_SAVE,
-                                        plugin_type=constants.IMPORTER_TYPE_ID,
-                                        working_dir=working_dir)
-        self.description = _('Saving images and tags')
+        return iter(self.parent.step_get_local_units.units_to_download)
 
-    def process_main(self):
+    def process_main(self, item=None):
         """
         Gets an iterable of units that were downloaded from the parent step,
         moves their files into permanent storage, and then saves the unit into
         the database and into the repository.
+
+        This method will be called exactly once without an item argument if the iterator returned by
```

Comment:
Same comment as before. Although you don't have the ``if`` block protection in this case, the comment can at least be removed.